hazelcast/src/main/java/com/hazelcast/instance/GroupProperties.java

```diff
@@ -214,6 +214,17 @@
      */
     public static final String PROP_JCACHE_PROVIDER_TYPE = "hazelcast.jcache.provider.type";
 
+    /**
+     * Result size limit of a predicate query operation.
```

Comment:
Can you provide a bit more information? E.g is this in total? Per member? Per partition? And why do we want this? We don't we get all results in one go? This is going to be asked by people. 

Also explain how it can be disabled so that HZ is working in old (but potentially dangerous) mode.